src/lib/starRelease.js

```diff
@@ -65,21 +65,36 @@ function generateLinearWithdrawTxs(contracts, amount, to) {
 }
 
 export async function getConditional(contracts, address) {
-  const [commitment, batches] = await Promise.all([
+  const [commitment, batches, remaining] = await Promise.all([
     conditionalSR.getCommitment(contracts, address),
     conditionalSR.getBatches(contracts, address),
+    conditionalSR.getRemainingStars(contracts, address),
   ]);
 
   if (commitment === null || batches === null) {
     return { total: 0, available: 0, withdrawn: 0, batchLimits: [] };
   }
 
-  const { total } = commitment;
-  const withdrawLimits = await Promise.all(
-    batches.map((_, idx) =>
-      conditionalSR.getWithdrawLimit(contracts, address, idx)
-    )
-  );
+  const forfeited = await conditionalSR.getForfeited(contracts, address);
+  const total = remaining.length;
+  let balance = total;
+
+  const withdrawLimits = await batches.reduce(async (p, _, idx) => {
+    const acc = await p;
```

Comment:
What is `acc` supposed to stand for? Something like `batchSize` would be much more readable. (Also why do we assume they're promises? Shouldn't `batches` be an array of numbers?)

Reply:
acc -> accumulator.
the product of batches.reduce is a promise returning an array of numbers.

Reply:
Oops, I was reading this as it it were a `.map`.